azurerm/internal/services/bot/bot_channel_skype_resource.go

```diff
@@ -0,0 +1,245 @@
+package bot
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/botservice/mgmt/2021-03-01/botservice"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/location"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/bot/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/bot/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/pluginsdk"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceBotChannelSkype() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceBotChannelSkypeCreate,
+		Read:   resourceBotChannelSkypeRead,
+		Delete: resourceBotChannelSkypeDelete,
+		Update: resourceBotChannelSkypeUpdate,
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.BotChannelID(id)
+			return err
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"bot_name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.BotName,
+			},
+
+			// issue: https://github.com/Azure/azure-rest-api-specs/issues/15170
+			// this field could not update to empty, so add `Computed: true` to avoid diff
+			"calling_web_hook": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validate.BotCallingWebHook(),
+			},
+
+			"enable_calling": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+			},
+
+			"enable_groups": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+			},
+
+			"enable_media_cards": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+			},
+
+			"enable_messaging": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+			},
```

Comment:
could we make this consistent with other bools?
```suggestion
			"calling_enabled": {
				Type:     pluginsdk.TypeBool,
				Optional: true,
			},

			"groups_enabled": {
				Type:     pluginsdk.TypeBool,
				Optional: true,
			},

			"media_cards_enabled": {
				Type:     pluginsdk.TypeBool,
				Optional: true,
			},

			"messaging_enabled": {
				Type:     pluginsdk.TypeBool,
				Optional: true,
			},
```